onnxruntime/python/tools/transformers/microbench/benchmark.py

```diff
@@ -0,0 +1,30 @@
+import time
+import onnxruntime as ort
+
+
+def create_session(onnx_file, provider, profiling=False):
+    sess_opt = ort.SessionOptions()
+    sess_opt.enable_profiling = profiling
+    if provider == "rocm":
+        execution_provider = ["ROCMExecutionProvider"] 
+    else:
+        execution_provider = ["CUDAExecutionProvider"] 
+        
+    sess = ort.InferenceSession(onnx_file, sess_options=sess_opt, providers=execution_provider)
+    return sess
+ 
+
+def benchmark(sess, io_binding):
+    # warm up    
+    for iter in range(10):
+      sess.run_with_iobinding(io_binding)    
+    
+    # measure 
+    max_iters = 100
+    start_time = time.time()
+    for iter in range(max_iters):
+        sess.run_with_iobinding(io_binding)    
```

Comment:
this is running the session with fixed inputs (generated ahead of time), if there is some model that perf is data dependent, people might want to benchmark with a given dataset. you might want to make it more flexible.

Reply:
Separated data generation from the io binding.